doc/source/ray-job-submission/overview.rst

```diff
@@ -0,0 +1,243 @@
+
+- **Job Manager**: An entity that manages the lifecycle of a Job and potentially multiple Ray clusters, such as scheduling, killing, polling status, getting logs, and persisting inputs / outputs. Should be highly available (HA) by default. Can be any framework with these abilities, such as Airflow.
+
+Within the context of Job submission, the packaging part is handled by :ref:`Runtime Environments<runtime-environments>`, where we can dynamically configure your desired Ray cluster environment, actor or task level runtime environment for your submitted Job.
+
+Example - Setup
+---------------
+
+Let's start with a sample Ray script as example for Job submission. Once executed locally, this script will use Ray APIs to print counter value of a remote actor from 1 to 5, and print the version of 'requests' module it's using.
+
+We can put this file in a local directory of your choice, with filename "script.py", so your working directory will look like:
+
+.. code-block:: bash
+
+  | your_working_directory ("./")
+  | ├── script.py
+
+.. code-block:: python
+
+    import ray
+    import requests
+
+    ray.init()
+
+    @ray.remote
+    class Counter:
+        def __init__(self):
+            self.counter = 0
+
+        def inc(self):
+            self.counter += 1
+
+        def get_counter(self):
+            return self.counter
+
+    counter = Counter.remote()
+
+    for _ in range(5):
+        ray.get(counter.inc.remote())
+        print(ray.get(counter.get_counter.remote()))
+
+    print(requests.__version__)
+
+
+| Ensure we have a local Ray cluster with a running head node. The address and port shown in terminal should be where we submit Job requests to.
+
+.. code-block:: bash
+
+   ❯ ray start --head
+    Local node IP: 127.0.0.1
+    INFO services.py:1360 -- View the Ray dashboard at http://127.0.0.1:8265
+
+Ray Job APIs
+------------
+
+We provide three APIs for Job submission: SDK, CLI and HTTP. Both SDK and CLI uses the same HTTP endpoints under the hood, where SDK provides programmatic submission and CLI facilitates command line development.
+
+**Key inputs to Job submission**
+
+- **Entrypoint**: Shell command to execute once your code is unpackaged with runtime environment configured
+    - Typically :code:`python your_script.py`, can also be any shell script such as :code:`echo hello`.
+- **Runtime Environment**:
+    - :code:`working_dir` as local directory: It will be automatically zipped and uploaded to the target Ray cluster, then unpacked to where your submitted application runs.
+    - :code:`working_dir` as remote URIs, such as S3, Git or others: It will be downloaded and unpacked to where your submitted application runs. For details, see :ref:`Runtime Environments<runtime-environments>`.
+
+Job CLI API
+-----------
+
+The easiest way to get started is to use Job submission CLI.
+
+If we have :code:`RAY_ADDRESS` environment variable set with a local Ray cluster:
```

Comment:
instead of this, just explicitly do `export RAY_ADDRESS=...` so people can copy-paste